include/ibamr/LevelSetUtilities.h

```diff
@@ -66,360 +66,392 @@ namespace IBAMR
-        {
-            return;
-        } // TagLevelSetRefinementCells
-
-        void setTagMinValue(double tag_min)
-        {
-            d_tag_min_value = tag_min;
-            return;
-        } // setTagMinValue
-
-        double getTagMinValue() const
-        {
-            return d_tag_min_value;
-        } // getTagMinValue
-
-        void setTagMaxValue(double tag_max)
-        {
-            d_tag_max_value = tag_max;
-            return;
-        } // setTagMaxValue
-
-        double getTagMaxValue() const
-        {
-            return d_tag_max_value;
-        } // getTagMaxValue
-
-    private:
-        double d_tag_min_value = 0.0;
-        double d_tag_max_value = 0.0;
-    }; // TagLevelSetRefinementCells
-
-    static void TagLSCells(SAMRAI::tbox::Pointer<SAMRAI::hier::BasePatchHierarchy<NDIM> > hierarchy,
-                           const int level_number,
-                           const double error_data_time,
-                           const int tag_index,
-                           const bool initial_time,
-                           const bool uses_richardson_extrapolation_too,
-                           void* ctx);
+        d_ncells = ncells;
+    } // setInterfaceHalfWidth
 
+    double getInterfaceHalfWidth() const
+    {
+        return d_ncells;
+    } // getInterfaceHalfWidth
+
+    SAMRAI::tbox::Pointer<AdvDiffHierarchyIntegrator> getAdvDiffHierarchyIntegrator() const
+    {
+        return d_adv_diff_integrator;
+    } // getAdvDiffHierarchyIntegrator
+
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > getLevelSetVariable(int idx = 0) const
+    {
+        return d_ls_vars[idx];
+    } // getLSVariable
+
+private:
+    SAMRAI::tbox::Pointer<AdvDiffHierarchyIntegrator> d_adv_diff_integrator;
+    std::vector<SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > > d_ls_vars;
+    double d_ncells = 1.0;
+};
+
+/*!
+ * \brief A lightweight class to tag grid cells containing the level set variable for grid refinement
+ */
+class TagLSRefinementCells
+{
+public:
     /*!
-     * \brief A lightweight class that stores the current value of the Lagrange multiplier
-     *  for the level set variable.
+     * \brief Constructor of the class.
      */
-    class LevelSetMassLossFixer : public LevelSetContainer, public SAMRAI::tbox::Serializable
+    TagLSRefinementCells(SAMRAI::tbox::Pointer<AdvDiffHierarchyIntegrator> adv_diff_integrator,
+                         SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > ls_var,
+                         double tag_min_value = 0.0,
+                         double tag_max_value = 0.0)
+        : d_ls_container(adv_diff_integrator, ls_var), d_tag_min_value(tag_min_value), d_tag_max_value(tag_max_value)
+    {
+        // intentionally left blank
+    } // TagLevelSetRefinementCells
+
+    const LevelSetContainer& getLevelSetContainer() const
+    {
+        return d_ls_container;
+    } // getLevelSetContainer
+
```

Comment:
There should be a function that return a non-const LevelSetContainer& in this class as well for consistency.